bd-logger/src/async_log_buffer.rs

```diff
@@ -447,14 +447,24 @@ impl<R: LogReplay + Send + 'static> AsyncLogBuffer<R> {
   async fn process_log(&mut self, log: LogLine, block: bool) -> anyhow::Result<Vec<Log>> {
     // Prevent re-entrancy when we are evaluating the log metadata.
     let result = with_thread_local_logger_guard(|| {
-      self
-        .metadata_collector
-        .normalized_metadata_with_extra_fields(
-          log.fields,
-          log.matching_fields,
-          log.log_type,
-          &mut self.global_state_tracker,
-        )
+      if let Some(LogAttributesOverrides::PreviousRunSessionID(_id, _timestamp)) =
```

Comment:
Maybe drop a note on the PreviousRunSessionID enum about this behavior? The interface doesn't make it obvious that the way fields are handled changes so might be good to document it at least